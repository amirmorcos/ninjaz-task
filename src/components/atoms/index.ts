import Icon from "./Icon/Icon";
import Divider from "./Divider/Divider";
import Avatar from "./Avatar/Avatar";

export { Icon, Divider, Avatar };
