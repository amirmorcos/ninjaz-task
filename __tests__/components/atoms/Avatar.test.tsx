import React from "react";
import renderer from "react-test-renderer";
import { Avatar } from "../../../src/components/atoms/index";

jest.mock("react-native-vector-icons/Ionicons", () => "Icon");
jest.mock("react-native-toast-message", () => ({
  show: jest.fn(),
  hide: jest.fn(),
}));
jest.mock("react-native-size-matters", () => ({
  scale: () => {},
  verticalScale: () => {},
  moderateScale: () => {},
}));

test("renders correctly", () => {
  const tree = renderer
    .create(
      <Avatar
        firstName="Amir"
        image="https://randomuser.me/api/portraits/med/women/33.jpg"
        lastName="Amir"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
