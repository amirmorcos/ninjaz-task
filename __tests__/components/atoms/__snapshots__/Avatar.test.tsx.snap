// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders  correctly 1`] = `
<View>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": undefined,
          "height": undefined,
          "width": undefined,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": undefined,
          "height": undefined,
          "width": undefined,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
